sszgen/hash.go

```diff
@@ -95,78 +95,112 @@ func (v *Value) hashRoots(isList bool, elem Type) string {
 	})
 }
 
-func (v *Value) hashTreeRoot() string {
+// takes a "name" param so that the variable name can be replaced with a local name
+// ie within a for loop for a list, the we want to refer to "elem" w/o a receiver variable
+// when not specified, name will be set to "::." + v.name. In the final templating pass,
+// the output formatter replaces all instances of "::" with the receiver variable for the container.
+func (v *Value) hashTreeRoot(name string, appendBytes bool) string {
+	if name == "" {
+		name = "::." + v.name
```

Comment:
consolidating the name string handling here so that an explicit name can be passed in. This is needed when the value is nested, ie in those cases the containing list will pass in the string `elem` for the `name` param.